Zend/zend_exceptions.c

```diff
@@ -758,6 +760,179 @@ static void declare_exception_properties(zend_class_entry *ce)
 		(zend_type) ZEND_TYPE_INIT_CE(zend_ce_throwable, /* allow_null */ 1, 0));
 }
 
+static void declare_stack_frame_properties(zend_class_entry *ce)
+{
+	zval val;
+	zend_string *prop;
+
+	ZVAL_NULL(&val);
```

Comment:
I forget exactly how you declare a typed property that is initially undeclared. Maybe `ZVAL_UNDEF(&val)`, but it might be more useful to just add MAY_BE_NULL to almost all of the ZEND_TYPE_INIT_MASK values.

In any case, this is initializing the property with a default that's intended to be impossible for unserialization, constructors, php code, etc. to set. This may cause problems with the engine or opcache if opcache ever supports inferences on typed properties? Things taking references to the properties may also behave unexpectedly. (e.g. with `--enable-debug` validating typed properties actually have the correct types)

```
php > array_map(fn() => $GLOBALS['x'] = StackFrame::getTrace()[0], [2]);
php > var_export($x);
StackFrame::__set_state(array(
   'file' => NULL,
   'line' => NULL,
   'function' => '{closure}',
   'class' => NULL,
   'type' => NULL,
   'object' => NULL,
   'object_class' => NULL,
   'closure' => NULL,
   'args' => 
  array (
    0 => 2,
  ),
))
php > var_export($x->file);
NULL
php > $x->file = 'file.php';
php > $x->file = null;

Warning: Uncaught TypeError: Cannot assign null to property StackFrame::$file of type string in php shell code:1
Stack trace:
#0 {main}
  thrown in php shell code on line 1
```